jsk_interactive_markers/jsk_interactive_marker/src/marker_6dof.cpp

```diff
@@ -242,7 +244,9 @@ class Marker6DOF {
 
   void publishTF(const geometry_msgs::PoseStamped& pose) {
     tf::Transform transform;
-    tf::poseMsgToTF(pose.pose, transform);
+    geometry_msgs::PoseStamped transformed_pose;
+    tf_listener_->transformPose(frame_id_, ros::Time(0), pose, pose.header.frame_id, transformed_pose);
+    tf::poseMsgToTF(transformed_pose.pose, transform);
```

Comment:
please try following example, the tf jumps just after we clicked the merker, the reason is IM publishes with respect to tha global frame defined in rviz and marker_6dof want to publish tf defined in rosparam. if the global frame and marker_6dof is not identical, I think we have problem. 

```
<launch>
  <node pkg="tf" type="static_transform_publisher"
        name="map_to_head_frame"
        args="0 0 1 0 0 0 map head_link 100" />
  <node pkg="tf" type="static_transform_publisher"
        name="camera_to_rgb_frame"
        args="0 0 1 0 0 0 camera_link rgb_link 100" />
  <node name="head_to_camera_frame"
        pkg="jsk_interactive_marker" type="marker_6dof">
    <rosparam>
frame_id: head_link
initial_orientation: [0.0, 0.0, 0.0, 1.0]
initial_x: 0.0
initial_y: 0.0
initial_z: 0.0
object_a: 0.2
object_b: 0.2
object_g: 0.0
object_r: 0.0
object_type: sphere
object_x: 0.1
object_y: 0.1
object_z: 0.1
publish_tf: true
tf_frame: camera_link
    </rosparam>
  </node>
  <node pkg="rviz" name="rviz" type="rviz" />
</launch>

```